frontend/src/components/post/EditPostModal.jsx

```diff
@@ -0,0 +1,275 @@
+    }
+
+    getBase64(e) {
+        var file = e.target.files[0];
+        if (file === undefined) {
+          return;
+        } else {
+          let reader = new FileReader();
+          reader.readAsDataURL(file);
+          reader.onload = () => {
+            this.setState(({image:reader.result}))
+          };
+          reader.onerror = (err) => {
+            console.log(err);
+            this.setState({msg: err}); 
+          };
+        }
+      }
+    
+    getImageUrl = async (e)=>  {
+        console.log(e.target.value)
+        this.setState(({image:e.target.value}))
+    }
+    
+    jpegFormItem = () => (
+        
+        <Form.Item 
+            name="content"
+            rules={[{message: "Please Upload An Image"}]}
+        >
+            <Radio.Group onChange={e => this.imageFromChange(e)} name="imageFrom">
+                <Radio.Button value="local">Local Input</Radio.Button>
+                <Radio.Button value="link">Link To Images</Radio.Button>
+            </Radio.Group>
+            {this.state.imageFrom === "local"?
+            <input required
+            type="file"
+            accept=".jpeg"
+            onChange={(e) => this.getBase64(e)}
+            />:null}
+            {this.state.imageFrom === "link"?
+            <input required 
+            type="url"
+            onChange={(e) => this.getImageUrl(e)}
+            />:null}
+            
+        </Form.Item>
+    );
+    
+    
+    pngFormItem = () => (
+        
+        <Form.Item 
+            name="content"
+            rules={[{ message: "Please Upload An Image"}]}
+        >
+        
+
+            <Radio.Group onChange={e => this.imageFromChange(e)} name="imageFrom">
+                <Radio.Button value="local">Local Input</Radio.Button>
+                <Radio.Button value="link">Link To Images</Radio.Button>
+            </Radio.Group>
+            {this.state.imageFrom === "local"?
+            <input required
+            type="file"
+            accept=".png"
+            onChange={(e) => this.getBase64(e)}
+            />:null}
+            {this.state.imageFrom === "link"?
+            <input required 
+            type="url"
+            onChange={(e) => this.getImageUrl(e)}
+            />:null}
+        
+        </Form.Item>
+    );
+
+    showModal = () => {
+        this.setState({visible: true});
+    }
+
+    onFinish = (values) => {
+        // It can be the case that the user provides no categories, in that case, default to []
+        if (values.categories === undefined) {
+            values.categories = []
+        }
+        if (values.unlisted === undefined) {
+            values.unlisted = false;
+        }
+        if (values.contentType==="image/png;base64" || values.contentType==="image/jpeg;base64"){
+            values.content = this.state.image;
+        }
+        this.props.editPost(values, this.props.initialValues.id); 
+        window.location.reload(false)
```

Comment:
This is bad practice to force a reload, you should change the state and update the post you are changing